### PR TITLE
Add Manage API Examples and Fix Formatting in Other Examples

### PR DESCRIPTION
**Pull request by dvonthenen**
Addresses issue: https://github.com/deepgram/deepgram-dotnet-sdk/issues/192

This introduces examples for the Manage APIs and includes CRUD examples for:
- balances
- invitations
- keys
- members
- projects
- scopes
- usage

The intended flow for the example applications follows CRUD:
1. list all objects (this would be keys or members or etc)
2. create an object
3. get the created object to verify it was successful
4. delete the object
5. list all objects to make sure it was deleted

Not all of these examples can follow this flow exactly because the operation may be destructive. In that case, the example with either excit early with a warning OR show you how to call the function as non-active AKA commented out code.

## Summary by CodeRabbit

- **New Features**
	- Introduced several new projects for enhanced functionality, including debugging, streaming, and management of invitations, keys, members, projects, usage, and scopes.
	- Enhanced JSON serialization options across various examples for improved readability and customization of output format.
	- Added functionality for managing invitations, keys, members, projects, scopes, and usage summaries using the Deepgram .NET SDK.
- **Bug Fixes**
	- Fixed a missing semicolon in the `Channel.cs` file.
	- Adjusted output formatting in streaming examples for clarity and consistency.
- **Refactor**
	- Updated data types and removed default values in several models to better align with usage requirements and flexibility.
- **Style**
	- Improved error handling and output messages in balance management examples for clearer user feedback.

### CODE REVIEW

File: examples/prerecorded/topic/Program.cs [hunk 1]
```diff
@@ -16,6 +16,12 @@ static async Task Main(string[] args)
             // Normal logging is "Info" level
             Library.Initialize();
 
+            // JSON options
+            JsonSerializerOptions options = new(JsonSerializerDefaults.Web)
+            {
+                WriteIndented = true
+            };
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/prerecorded/topic/Program.cs:23]: These examples are designed to be self-contained. We don't want to centralize the JSON parsing options among the various examples.

Dismissing this comment.

File: examples/prerecorded/intent/Program.cs [hunk 2]
```diff
@@ -16,6 +16,12 @@ static async Task Main(string[] args)
             // Normal logging is "Info" level
             Library.Initialize();
 
+            // JSON options
+            JsonSerializerOptions options = new(JsonSerializerDefaults.Web)
+            {
+                WriteIndented = true
+            };
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/prerecorded/intent/Program.cs:23]: These examples are designed to be self-contained. We don't want to centralize the JSON parsing options among the various examples.

Dismissing this comment.

File: examples/prerecorded/sentiment/Program.cs [hunk 3]
```diff
@@ -16,6 +16,12 @@ static async Task Main(string[] args)
             // Normal logging is "Info" level
             Library.Initialize();
 
+            // JSON options
+            JsonSerializerOptions options = new(JsonSerializerDefaults.Web)
+            {
+                WriteIndented = true
+            };
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/prerecorded/sentiment/Program.cs:23]: These examples are designed to be self-contained. We don't want to centralize the JSON parsing options among the various examples.

Dismissing this comment.

File: examples/prerecorded/file/Program.cs [hunk 4]
```diff
@@ -19,6 +19,12 @@
             // OR very chatty logging
             //Library.Initialize(LogLevel.Debug); // LogLevel.Default, LogLevel.Debug, LogLevel.Verbose
 
+            // JSON options
+            JsonSerializerOptions options = new(JsonSerializerDefaults.Web)
+            {
+                WriteIndented = true
+            };
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/prerecorded/file/Program.cs:26]: These examples are designed to be self-contained. We don't want to centralize the JSON parsing options among the various examples.

Dismissing this comment.

File: examples/manage/invitations/Program.cs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+            // Set "DEEPGRAM_API_KEY" environment variable to your Deepgram API Key
+            var deepgramClient = new ManageClient();
+
+            // get projects
+            var projectResp = await deepgramClient.GetProjects();
+            if (projectResp == null)
+            {
+                Console.WriteLine("ListProjects failed.");
+                Environment.Exit(1);
+            }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/invitations/Program.cs:37]: These examples are designed to be self-contained. We don't want to centralize the JSON parsing options among the various examples.

Dismissing this comment.

File: examples/manage/members/Program.cs [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+            // Set "DEEPGRAM_API_KEY" environment variable to your Deepgram API Key
+            var deepgramClient = new ManageClient();
+
+            // Get projects
+            var projectResp = deepgramClient.GetProjects();
+            if (projectResp == null)
+            {
+                Console.WriteLine("ListProjects failed.");
+                Environment.Exit(1);
+            }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/members/Program.cs:39]: These examples are designed to be self-contained. We don't want to centralize the JSON parsing options among the various examples.

Dismissing this comment.

File: examples/prerecorded/file/Program.cs [hunk 7]
```diff
@@ -3,7 +3,7 @@
 // SPDX-License-Identifier: MIT
 
 using System.Text.Json;
-
+using System.Xml;
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/prerecorded/file/Program.cs:6]: will do!

File: examples/manage/balances/Program.cs [hunk 8]
```diff
@@ -28,48 +34,41 @@
...
-            //Console.WriteLine("\n\nBalances:");
-            //Console.WriteLine(JsonSerializer.Serialize(balanacesResponse));
-            //Console.WriteLine("\n\n");
+            var balanaceResponse = deepgramClient.GetBalance(projectId, balanceId);
+            if (balanaceResponse == null)
+            {
+                Console.WriteLine("\n\nNo balance found.\n\n");
+                return;
+            }
+            Console.WriteLine($"\n\n{JsonSerializer.Serialize(balanacesResponse, options)}\n\n");
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/balances/Program.cs:69]: will do

File: examples/manage/balances/Program.cs [hunk 9]
```diff
@@ -19,6 +19,12 @@ static async Task Main(string[] args)
             // OR very chatty logging
             //Library.Initialize(LogLevel.Debug); // LogLevel.Default, LogLevel.Debug, LogLevel.Verbose
 
+            // JSON options
+            JsonSerializerOptions options = new(JsonSerializerDefaults.Web)
+            {
+                WriteIndented = true
+            };
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/balances/Program.cs:26]: These examples are designed to be self-contained. We don't want to centralize the JSON parsing options among the various examples.

Dismissing this comment.

File: examples/manage/scopes/Program.cs [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+    {
+        private const string MEMBER_BY_EMAIL = "MEMBER_TO_DELETE_BY_EMAIL";
+
+        static async Task Main(string[] args)
+        {
+            // Initialize Library with default logging
+            // Normal logging is "Info" level
+            Library.Initialize();
+            // OR very chatty logging
+            //Library.Initialize(LogLevel.Debug); // LogLevel.Default, LogLevel.Debug, LogLevel.Verbose
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/scopes/Program.cs:22]: This was intentionally left behind in order to easily switch between normal and debug logging.

File: examples/manage/keys/Program.cs [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+{
+    class Program
+    {
+        static async Task Main(string[] args)
+        {
+            // Initialize Library with default logging
+            // Normal logging is "Info" level
+            Library.Initialize();
+            // OR very chatty logging
+            //Library.Initialize(LogLevel.Debug); // LogLevel.Default, LogLevel.Debug, LogLevel.Verbose
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/keys/Program.cs:20]: This was intentionally left behind in order to easily switch between normal and debug logging. Dismissing suggestion.

File: examples/manage/members/Program.cs [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+    {
+        private const string DELETE_MEMBER_BY_EMAIL = "MY_EMAIL";
+
+        static async Task Main(string[] args)
+        {
+            // Initialize Library with default logging
+            // Normal logging is "Info" level
+            Library.Initialize();
+            // OR very chatty logging
+            //Library.Initialize(LogLevel.Debug); // LogLevel.Default, LogLevel.Debug, LogLevel.Verbose
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/members/Program.cs:22]: This was intentionally left behind in order to easily switch between normal and debug logging. Dismissing suggestion.

File: examples/manage/invitations/Program.cs [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+{
+    class Program
+    {
+        static async Task Main(string[] args)
+        {
+            // Initialize Library with default logging
+            // Normal logging is "Info" level
+            Library.Initialize();
+            // OR very chatty logging
+            //Library.Initialize(LogLevel.Debug); // LogLevel.Default, LogLevel.Debug, LogLevel.Verbose
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/invitations/Program.cs:20]: This was intentionally left behind in order to easily switch between normal and debug logging. Dismissing suggestion.

File: examples/manage/invitations/Program.cs [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+            {
+                Console.WriteLine($"\n\n{JsonSerializer.Serialize(listResp, options)}\n\n");
+            }
+
+            // send invite
+            var createInvite = new InviteSchema()
+            {
+                Email = "spam@spam.com",
+                Scope = "member",
+            };
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/invitations/Program.cs:65]: This is just using a default which is fine in this case. Dismissing suggestion.

File: examples/manage/members/Program.cs [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+            {
+                Console.WriteLine($"\n\n{JsonSerializer.Serialize(listResp, options)}\n\n");
+
+                foreach (var member in listResp.Result.Members)
+                {
+                    if (member.Email == DELETE_MEMBER_BY_EMAIL)
+                    {
+                        delMemberId = member.MemberId;
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/members/Program.cs:66]: A user is going to run the example locally in which they will need to update the required fields. Dismissing suggestion.

File: examples/manage/scopes/Program.cs [hunk 16]
```diff
@@ -0,0 +1,130 @@
...
+            else
+            {
+                Console.WriteLine($"\n\n{JsonSerializer.Serialize(listResp, options)}\n\n");
+
+                foreach (var member in listResp.Members)
+                {
+                    if (member.Email == MEMBER_BY_EMAIL)
+                    {
+                        memberId = member.MemberId;
+                    }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/scopes/Program.cs:67]: A user is going to run the example locally in which they will need to update the required fields. Dismissing suggestion.

File: examples/manage/projects/Program.cs [hunk 17]
```diff
@@ -0,0 +1,130 @@
...
+            string myDeleteId = null;
+            foreach (var project in listResp.Projects)
+            {
+                if (project.Name == DELETE_PROJECT_BY_NAME)
+                {
+                    myDeleteId = project.ProjectId;
+                }
+                myId = project.ProjectId;
+                myName = project.Name;
+            }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/projects/Program.cs:52]: A user is going to run the example locally in which they will need to update the required fields. Dismissing suggestion.

File: examples/manage/scopes/Program.cs [hunk 18]
```diff
@@ -0,0 +1,130 @@
...
+
+            // Set "DEEPGRAM_API_KEY" environment variable to your Deepgram API Key
+            var deepgramClient = new ManageClient();
+
+            // get projects
+            var projectResp = await deepgramClient.GetProjects();
+            if (projectResp == null)
+            {
+                Console.WriteLine("ListProjects failed.");
+                Environment.Exit(1);
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/scopes/Program.cs:38]: A user is going to run the example locally in order to see the behavior of this example. They will know when and why it failed. Dismissing suggestion.

File: examples/manage/usage/Program.cs [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+                Console.WriteLine($"\n\n{JsonSerializer.Serialize(fieldsResp, options)}\n\n");
+            }
+
+            // list usage
+            var summaryOptions = new UsageSummarySchema();
+            var summaryResp = await deepgramClient.GetUsageSummary(myId, summaryOptions);
+            if (summaryResp == null)
+            {
+                Console.WriteLine("UsageSummary not found");
+            }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/usage/Program.cs:99]: A user is going to run the example locally in order to see the behavior of this example. They will know when and why it failed. Dismissing suggestion.

File: examples/manage/usage/Program.cs [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+                Console.WriteLine($"\n\n{JsonSerializer.Serialize(reqResp, options)}\n\n");
+            }
+
+            // get fields
+            var fieldsOptions = new UsageFieldsSchema();
+            var fieldsResp = await deepgramClient.GetUsageFields(myId, fieldsOptions);
+            if (fieldsResp == null)
+            {
+                Console.WriteLine("UsageFields not found.");
+                Environment.Exit(1);
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/usage/Program.cs:86]: A user is going to run the example locally in order to see the behavior of this example. They will know when and why it failed. Dismissing suggestion.

File: examples/manage/usage/Program.cs [hunk 21]
```diff
@@ -0,0 +1,113 @@
...
+            Console.WriteLine($"\n\n{JsonSerializer.Serialize(projectResp, options)}\n\n");
+
+            // list requests
+            string requestId = null;
+            var requestsOptions = new UsageRequestsSchema();
+            var listResp = await deepgramClient.GetUsageRequests(myId, requestsOptions);
+            if (listResp == null)
+            {
+                Console.WriteLine("\n\nNo requests found\n\n");
+            }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/usage/Program.cs:56]: A user is going to run the example locally in order to see the behavior of this example. They will know when and why it failed. Dismissing suggestion.

File: examples/manage/usage/Program.cs [hunk 22]
```diff
@@ -0,0 +1,113 @@
...
+            // Set "DEEPGRAM_API_KEY" environment variable to your Deepgram API Key
+            var deepgramClient = new ManageClient();
+
+            // get projects
+            var projectResp = await deepgramClient.GetProjects();
+            if (projectResp == null)
+            {
+                Console.WriteLine("ListProjects failed.");
+                Environment.Exit(1);
+            }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/usage/Program.cs:37]: A user is going to run the example locally in order to see the behavior of this example. They will know when and why it failed. Dismissing suggestion.

**Review by lukeocodes — APPROVED** [review]

**Pull request merged**